osu.Game.Tests/Visual/Online/TestSceneOfflineCommentsContainer.cs

```diff
@@ -3,90 +3,52 @@
+    public class TestSceneOfflineCommentsContainer : OsuTestScene
     {
         public override IReadOnlyList<Type> RequiredTypes => new[]
         {
+            typeof(CommentsContainer),
+            typeof(CommentsHeader),
             typeof(DrawableComment),
-            typeof(CommentsPage),
+            typeof(HeaderButton),
+            typeof(OverlaySortTabControl<>),
+            typeof(ShowChildrenButton),
+            typeof(DeletedCommentsCounter),
+            typeof(VotePill)
         };
 
         [Cached]
-        private readonly OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Purple);
-
-        private readonly BindableBool showDeleted = new BindableBool();
-        private readonly Container content;
+        private readonly OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Blue);
 
-        private TestCommentsPage commentsPage;
+        private TestCommentsContainer comments;
 
-        public TestSceneCommentsPage()
+        [SetUp]
+        public void SetUp() => Schedule(() =>
         {
-            Add(new FillFlowContainer
+            Clear();
+            Add(new BasicScrollContainer
             {
-                AutoSizeAxes = Axes.Y,
-                RelativeSizeAxes = Axes.X,
-                Direction = FillDirection.Vertical,
-                Spacing = new Vector2(0, 10),
-                Children = new Drawable[]
-                {
-                    new Container
-                    {
-                        AutoSizeAxes = Axes.Y,
-                        Width = 200,
-                        Child = new OsuCheckbox
-                        {
-                            Current = showDeleted,
-                            LabelText = @"Show Deleted"
-                        }
-                    },
-                    content = new Container
-                    {
-                        RelativeSizeAxes = Axes.X,
-                        AutoSizeAxes = Axes.Y,
-                    }
-                }
+                RelativeSizeAxes = Axes.Both,
+                Child = comments = new TestCommentsContainer()
             });
-        }
-
-        [Test]
-        public void TestAppendDuplicatedComment()
-        {
-            AddStep("Create page", () => createPage(getCommentBundle()));
-            AddAssert("Dictionary length is 10", () => commentsPage?.DictionaryLength == 10);
-            AddStep("Append existing comment", () => commentsPage?.AppendComments(getCommentSubBundle()));
-            AddAssert("Dictionary length is 10", () => commentsPage?.DictionaryLength == 10);
-        }
+        });
 
         [Test]
-        public void TestEmptyBundle()
+        public void TestLocalCommentBundle()
         {
-            AddStep("Create page", () => createPage(getEmptyCommentBundle()));
-            AddAssert("Dictionary length is 0", () => commentsPage?.DictionaryLength == 0);
-        }
```

Comment:
What was the point of having these tests if they're going to get removed with this refactoring? The first one in particular was a *regression test*. This is unacceptable.